cadc-registry/src/main/java/ca/nrc/cadc/reg/client/LocalAuthority.java

```diff
@@ -104,24 +106,46 @@ public LocalAuthority() {
 
         for (String std : mvp.keySet()) {
             List<String> values = mvp.getProperty(std);
-            if (values.size() > 1) {
-                throw new IllegalStateException("found " + values.size() + " values for " + std);
-            }
-            if (values.isEmpty()) {
-                log.debug(std + " has no value, skipping");
-            } else {
-                String val = values.get(0);
-                log.debug("authorityMap: " + std + " -> " + val);
-                authorityMap.put(std, val);
+            Set<URI> vals = new HashSet<>();
+            URI stdURI = URI.create(std);
+            authorityMap.put(stdURI, vals);
+            for (String val : values) {
+                URI valURI = URI.create(val);
+                log.debug("authorityMap: " + stdURI + " -> " + valURI);
+                vals.add(valURI);
             }
         }
     }
 
+    /**
+     * Returns the service URI associated with the baseStandardID. This method fails if the local authority is
+     * configured with more than one service URI corresponding to the baseStandard. Use `getServiceURIs` method for
+     * the more generic case
+     * @param baseStandardID base standard ID
+     * @return corresponding service URI (http or ivo)
+     * @deprecated deprecated in favour of getServiceURIs method
+     */
     public URI getServiceURI(String baseStandardID) {
-        String resourceIdentifier = authorityMap.get(baseStandardID);
-        if (resourceIdentifier == null) {
+        Set<URI> resourceIdentifiers = authorityMap.get(URI.create(baseStandardID));
+        if ((resourceIdentifiers == null) || (resourceIdentifiers.isEmpty())) {
+            throw new NoSuchElementException("not found: " + baseStandardID);
+        }
+        if (resourceIdentifiers.size() > 1) {
+            throw new NoSuchElementException("Multiple service URIs found for " + baseStandardID);
+        }
+        return resourceIdentifiers.iterator().next();
+    }
+
+    /**
+     * Returns the URIs of services associated with the baseStandardID.
+     * @param baseStandardID base standard ID URI
```

Comment:
I see that it was already called this, but I don't know why this is called `baseStandardID` rather than just `standardID`... the way we use it is the latter. We could have used _base_ in the sense that we could have used
```
ivo://ivoa.net/std/GMS = ivo://cadc.nrc.ca/gms
```
instead of an entry for each actual standardID in that spec, but this usage is not _base_:
```
ivo://ivoa.net/sso#OpenID = https://ska-iam.stfc.ac.uk/
```
and base isn't really usable with sso at all.

So right now _base_ is probably the wrong qualifier there and we'd have to write more careful code to make _base_ concept work (specifically, find keys in the map that start with the baseStandardID, rather than exact match).

So, change terminology??

Reply:
Agree